Mathlib/Data/PFunctor/Univariate/Basic.lean

```diff
@@ -136,11 +136,142 @@ theorem iget_map [DecidableEq P.A] [Inhabited α] [Inhabited β] (x : P α)
 end PFunctor
 
 /-
-Composition of polynomial functors.
+Constructions on polynomial functors.
 -/
 namespace PFunctor
 
-/-- Composition for polynomial functors -/
+section Basic
+
+/-- The zero polynomial functor, defined as `A = PEmpty` and `B _ = PEmpty` -/
+def zero : PFunctor.{uA, uB} := ⟨PEmpty, fun _ => PEmpty⟩
+
+/-- The unit polynomial functor, defined as `A = PUnit` and `B _ = PEmpty` -/
+def one : PFunctor.{uA, uB} := ⟨PUnit, fun _ => PEmpty⟩
+
+instance : Zero PFunctor.{uA, uB} where
+  zero := zero
+
+instance : One PFunctor.{uA, uB} where
+  one := one
```

Comment:
Here you're making things hard for yourself by having both the def and the instance. I recommend inlining the former into the latter, so that you can drop half the lemmas.